<!doctype html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

        <title>Long-read assembly of Norway spruce and Scots pine</title>

        <link rel="stylesheet" href="css/reveal.css">
        <link rel="stylesheet" href="css/theme/maehler_dark.css">

        <!-- Theme used for syntax highlighting of code -->
        <link rel="stylesheet" href="lib/css/zenburn.css">

        <!-- Font awesome -->
        <link rel="stylesheet" href="lib/css/fontawesome.css">
        <link rel="stylesheet" href="lib/css/solid.css">
        <link rel="stylesheet" href="lib/css/brands.css">

        <!-- Custom modifications that shouldn't go into the theme -->
        <link rel="stylesheet" href="css/customise.css">

        <!-- Printing and PDF exports -->
        <script>
            var link = document.createElement( 'link' );
            link.rel = 'stylesheet';
            link.type = 'text/css';
            link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
        </script>
    </head>
    <body>
        <div class="slidelink"><a href=""></a></div>
        <div class="twitterlink"><a href="https://twitter.com/nmaehler"><i class="fab fa-twitter"></i>@nmaehler</a></div>
        <div class="reveal">
            <div class="slides">
                <section>
                    <h2>Long-read assembly of Norway spruce and Scots pine</h2>
                    <p>
                    <small>
                        Niklas Mähler<br>
                        Uppsala Long-read sequencing meeting<br>
                        2019-04-03
                    </small>
                    </p>
                </section>

                <section>
                    <section>
                        <h2>Why spruce and pine?</h2>
                    </section>

                    <section>
                        <img src="img/virkesforrad.png"
                            alt="Virkesförråd"
                            style="background: white;">
                    </section>

                    <section>
                        <img src="img/handel.png"
                            alt="Most important categories for import/export"
                            style="background: white;">
                    </section>

                    <section>
                        <h2>The availability of field trials</h2>
                    </section>

                    <section data-background-image="img/spruce_trials_trim.png"
                             data-background-position="bottom 0 right 0"
                             data-background-size="35%">
                        <p style="width: 60%">Current spruce trials</p>
                    </section>

                    <section data-background-image="img/pine_trials_trim.png"
                             data-background-position="bottom 0 right 0"
                             data-background-size="35%">
                        <p style="width: 60%">Current pine trials</p>
                    </section>

                    <section>
                        <h3>Problems with traditional breeding</h3>
                        <p>Generation time for pine: 11&ndash;18 years</p>
                        <p>Generation time for spruce: ~30 years</p>
                    </section>

                    <section>
                        <h3>Decrease generation time with genomic selection</h3>
                    </section>
                </section>

                <section>
                    <section>
                        <h2>Why a new spruce assembly?</h2>
                        <figure>
                            <img src="img/nystedt2013.png" alt="Nystedt et al. 2013">
                            <figcaption>Nystedt et al. 2013. Nature, doi: <a href="https://doi.org/10.1038/nature12211">10.1038/nature12211</a>
                            </figcaption>
                        </figure>
                        <aside class="notes">
                            <p>
                            Still a huge achievement at the time given the existing
                            technology and that no other conifer had been sequenced.
                            </p>
                        </aside>
                    </section>

                    <section>
                        <h2>Spruce v1</h2>
                        <ul style="font-size: 0.8em;">
                            <li>More than 10 million scaffolds.</li>
                            <li>~30% of Trinity transcripts were split across scaffolds.</li>
                        </ul>
                        <figure style="margin-top: 1em;">
                            <img src="img/mads_box_fragmentation.png" alt="Fragmentation of MADS-box gene">
                            <figcaption>MADS-box gene mapping to three scaffolds</figcaption>
                        </figure>
                        <aside class="notes">
                            <p>Just the sheer number of scaffolds make a lot of downstream analyses difficult</p>
                            <p>Gene fragmentation causes issues for wet lab scientists, for example when trying
                            to amplify a gene with PCR.</p>
                            <p>Many of the fragmentations caused by repeats. With long reads we might be able
                            to better resolve these repeats and obtain a less fragmented assembly.</p>
                        </aside>
                    </section>
                </section>

                <section>
                    <div style="display: flex;">
                        <div style="display: block; width: 100%;">
                            <h2>Norway spruce</h2>
                            <ul>
                                <li><i>Picea abies</i></li>
                                <li>20 Gbp genome</li>
                                <li>~70-80% repeat content</li>
                                <li>~30-60,000 genes (?)</li>
                            </ul>
                        </div>
                        <div style="display: block; width: 100%;">
                            <h2>Scots pine</h2>
                            <ul>
                                <li><i>Pinus sylvestris</i></li>
                                <li>22.5 Gbp genome</li>
                                <li>? repeat content</li>
                                <li>? genes</li>
                            </ul>
                        </div>
                    </div>
                </section>

                <section>
                    <section>
                        <h1>Data</h1>
                        <aside class="notes">
                            From here on I will only talk about spruce since this
                            is what we currently have data for.
                        </aside>
                    </section>

                    <section>
                        <ul>
                            <li>
                                50X Pacbio Sequel genomic DNA
                                <ul>
                                    <li>Clone Z4006 from Ragunda kommun, Jämtland</li>
                                    <li>157 million reads</li>
                                    <li>Subread N50: 12 kbp</li>
                                </ul>
                            </li>
                            <li>HiC data coming</li>
                            <li>Chromium 10X coming</li>
                            <li>Some IsoSeq here, more coming</li>
                        </ul>
                    </section>

                    <section>
                        <h2>Contaminant filtering</h2>
                        <table>
                            <thead>
                                <tr>
                                    <th>Type</th>
                                    <th>Count</th>
                                    <th>Percentage</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>Total reads</td>
                                    <td>157,235,560</td>
                                    <td>100%</td>
                                </tr>
                                <tr>
                                    <td>Chloroplast</td>
                                    <td>903,359</td>
                                    <td>0.58%</td>
                                </tr>
                                <tr>
                                    <td>Mitochondria</td>
                                    <td>12,455,470</td>
                                    <td>7.9%</td>
                                </tr>
                            </tbody>
                        </table>
                        <figure style="top: 0;" class="acknowledgement-fig">
                            <img src="https://avatars2.githubusercontent.com/u/8308279?s=400&v=4"
                                alt="Bastian Schiffthaler">
                            <figcaption>Bastian Schiffthaler</figcaption>
                        </figure>
                    </section>

                    <section>
                        <h2>Downsampling</h2>
                        <p>Proceed with the longest 30X</p>
                        <ul>
                            <li>Shortest read: 8 kbp</li>
                            <li>N50: 15,105</li>
                        </ul>
                        <figure style="top: 0;" class="acknowledgement-fig">
                            <img src="https://avatars2.githubusercontent.com/u/8308279?s=400&v=4"
                                alt="Bastian Schiffthaler">
                            <figcaption>Bastian Schiffthaler</figcaption>
                        </figure>
                    </section>
                </section>

                <section>
                    <h1>Keyword: preliminary</h1>
                </section>

                <section>
                    <h1>Software</h1>
                    <p>
                    We have considered two assemblers:<br>
                    <strong>MARVEL</strong> and <strong>wtdbg2</strong>
                    </p>
                </section>

                <section>
                    <section>
                        <h1>MARVEL</h1>
                    </section>

                    <section>
                        <p>Developed for the 32 Gbp axolotl genome</p>
                        <figure>
                            <img src="http://www.h-its.org/wp-content/uploads/2018/01/2017_Axolotl_IMP.jpg" alt="Axolotl">
                            <figcaption>&copy; IMP</figcaption>
                        </figure>
                        <figure style="top: 3em; right: 3em; max-width: 17%;" class="acknowledgement-fig">
                            <img src="https://s14-eu5.startpage.com/cgi-bin/serveimage?url=https:%2F%2Fwww.h-its.org%2Fwp-content%2Fuploads%2F2017%2F02%2Fwww.keskin-arts.com-DSC_8987.jpg&sp=a6c0bf3573f3a473335b15caadf73b94" alt="Siegfried Schloissnig">
                            <figcaption>Siegfried Schloissnig</figcaption>
                        </figure>
                    </section>

                    <section>
                        <h2>Philosophy</h2>
                        <ul>
                            <li>All vs all alignments</li>
                            <li>No error correction</li>
                            <li>Dynamically mask repeats</li>
                        </ul>
                    </section>

                    <section>
                        <h2>Axolotl runs</h2>
                        <p>150,000 CPU hours for one alignment round</p>
                        <p>Without repeat masking: several million CPU hours and more than 2 PB of storage.</p>
                    </section>

                    <section>
                        <h2>What about spruce?</h2>
                        <p>About 2 million CPU hours and 1 PB of storage for one alignment round.</p>
                    </section>

                    <section>
                        <h2>What went wrong?</h2>
                        <p><strong>Technical</strong></p>
                        <ul>
                            <li>Hiccups with the masking server?</li>
                        </ul>
                        <p><strong>Biological</strong></p>
                        <ul>
                            <li>Repeats too diverged in spruce?</li>
                        </ul>
                    </section>
                </section>

                <section>
                    <section>
                        <h1>wtdbg2</h1>
                        <p><small>say that fast 5 times&hellip;</small></p>
                    </section>

                    <section>
                        <p>Developed by Jue Ruan and Heng Li</p>
                        <!-- TODO: Add screenshot of their preprint. -->
                        <figure>
                            <img src="img/ruan_li_2019.png" alt="Ruan and Li 2019">
                            <figcaption>Ruan and Li 2019. bioRiv, doi: <a href="https://doi.org/10.1101/530972">10.1101/530972</a></figcaption>
                        </figure>
                        <aside class="notes">
                            <p>Heng Li, famous for samtools, bwa-mem, minimap2, and so on.</p>
                        </aside>
                    </section>

                    <section>
                        <p style="width: 85%;">Assembly done on reads longer than 12 kbp with version 1.2.8</p>
                        <figure class="acknowledgement-fig">
                            <img src="http://www.igp.uu.se/digitalAssets/122/c_122555-l_3-k_ignasbunikis.jpg"
                                alt="Ignas Bunikis">
                            <figcaption>Ignas Bunikis</figcaption>
                        </figure>
                    </section>

                    <section>
                        <table style="font-size: 0.7em;">
                            <thead>
                                <tr>
                                    <th>Assembly</th>
                                    <th>Total assembly size</th>
                                    <th>NG50</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>Spruce v1</td>
                                    <td>12.3 Gbp</td>
                                    <td>0.606 kbp</td>
                                </tr>
                                <tr>
                                    <td>Spruce v3 wtdbg</td>
                                    <td>20.1 Gbp</td>
                                    <td>99.1 kbp</td>
                                </tr>
                                <tr>
                                    <td>Aspen v2 wtdbg2</td>
                                    <td>431 Mbp</td>
                                    <td>387 kbp</td>
                                </tr>
                            </tbody>
                        </table>
                    </section>

                    <section>
                        <h2>BUSCO analysis</h2>
                        <table style="font-size: 0.7em">
                            <thead>
                                <tr>
                                    <th>Assembly</th>
                                    <th>Complete</th>
                                    <th>Fragmented</th>
                                    <th>Missing</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr class="fragment">
                                    <td>Spruce v1</td>
                                    <td>573</td>
                                    <td>128</td>
                                    <td>739</td>
                                </tr>
                                <tr class="fragment">
                                    <td>Spruce v3 wtdbg</td>
                                    <td>26</td>
                                    <td>15</td>
                                    <td>1399</td>
                                </tr>
                                <tr class="fragment">
                                    <td>Aspen v2 wtdbg2</td>
                                    <td>1182</td>
                                    <td>69</td>
                                    <td>189</td>
                                </tr>
                            </tbody>
                        </table>
                        <figure class="acknowledgement-fig">
                            <img src="img/tomas_larsson.jpg" alt="Tomas Larsson">
                            <figcaption>Tomas Larsson</figcaption>
                        </figure>
                    </section>

                    <section>
                        <h2>IsoSeq alignments</h2>
                        <img class="stretch" 
                            src="img/isoseq_alignments.png"
                            alt="IsoSeq GMAP alignments">
                        <figure class="acknowledgement-fig">
                            <img src="img/nico.jpeg" alt="Nicolas Delhomme">
                            <figcaption>Nicolas Delhomme</figcaption>
                        </figure>
                    </section>
                </section>

                <section>
                    <h2>Summary</h2>
                </section>

                <section>
                    <h3>Acknowledgements</h3>
                    <div style="display: flex; flex-wrap: wrap; justify-content: space-around; align-content: space-between; align-items: center; font-size: 0.6em;">
                        <div class="acknowledgement">
                            <h4>Umeå</h4>
                            <ul>
                                <li>Nathaniel Street</li>
                                <li>Ove Nilsson</li>
                                <li>Harry Wu</li>
                                <li>Nicolas Delhomme</li>
                                <li>Bastian Schiffthaler</li>
                                <li>Alexis Sullivan</li>
                            </ul>
                        </div>

                        <div class="acknowledgement" style="width: 30%">
                            <figure><img src="img/umu-logo-neg-EN.png" alt="Umeå University"></figure>
                            <figure style="width: 40%;"><img src="img/slu_logo_vit.png" alt="SLU"></figure>
                            <figure style="width: 30%;"><img src="img/upsc_logo.png" alt="UPSC"></figure>
                        </div>

                        <div class="acknowledgement">
                            <figure>
                                <img style="width: 10em;" src="img/scilifelab_green_white.png" alt="SciLifeLab">
                            </figure>
                            <ul>
                                <li>Tomas Larsson</li>
                                <li>Manfred Grabherr</li>
                                <li>Ignas Bunikis</li>
                                <li>Remi-André Olsen</li>
                                <li>Ellen Sherwood</li>
                                <li>Ulf Gyllensten</li>
                                <li>Björn Nystedt</li>
                            </ul>
                        </div>

                        <div class="acknowledgement">
                            <h4>Vienna</h4>
                            <ul>
                                <li>Siegfried Schloissnig</li>
                            </ul>
                            <figure style="margin-top: 1em;">
                                <img style="width: 10em;" src="img/IMP_Logo.jpg" alt="IMP">
                            </figure>
                        </div>

                        <div class="acknowledgement">
                            <h4>Computing</h4>
                            <ul>
                                <li>UPPMAX</li>
                                <li>SNIC</li>
                                <li>Marcus Lundberg</li>
                            </ul>
                        </div>

                        <div class="acknowledgement">
                            <h4>Funding</h4>
                            <figure style="margin-top: 1em;">
                                <img style="width: 10em;" src="img/kaw_logo_white.svg" alt="KAW">
                            </figure>
                        </div>
                    </div>
                </section>
            </div>
        </div>

        <script src="lib/js/head.min.js"></script>
        <script src="js/reveal.js"></script>

        <script>
            // More info about config & dependencies:
            // - https://github.com/hakimel/reveal.js#configuration
            // - https://github.com/hakimel/reveal.js#dependencies
            Reveal.initialize({
                dependencies: [
                    { src: 'plugin/markdown/marked.js' },
                    { src: 'plugin/markdown/markdown.js' },
                    { src: 'plugin/notes/notes.js', async: true },
                    { src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } }
                ],
                history: true
            });
        </script>
    </body>
</html>
